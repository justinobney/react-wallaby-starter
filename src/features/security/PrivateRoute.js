import React, {createElement} from 'react';
import {Redirect, Route} from 'react-router-dom';
import {connect} from 'react-redux';

export const PrivateRoute = ({component, user, ...rest}) => {
  const canRoute = !!user;

  return (
    <Route
      {...rest}
      render={props =>
        canRoute ? (
          createElement(component, props)
        ) : (
          <Redirect
            to={{
              pathname: '/auth/login',
              state: {from: props.location},
            }}
          />
        )}
    />
  );
};

const mapStateToProps = state => {
  return {
    user: state.identity.user,
  };
};

export default connect(mapStateToProps)(PrivateRoute);
