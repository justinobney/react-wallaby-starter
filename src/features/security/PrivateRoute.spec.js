import React from 'react';
import {create} from 'react-test-renderer';
import {MemoryRouter} from 'react-router';
import {PrivateRoute} from './PrivateRoute';

const RouteComponent = () => <div />;

describe('Component: PrivateRoute', () => {
  it('should render component when user exists', () => {
    const tree = create(
      <MemoryRouter initialEntries={['/app']} initialIndex={1}>
        <PrivateRoute path="/app" user={{}} component={RouteComponent} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render null when user does not exist', () => {
    const tree = create(
      <MemoryRouter initialEntries={['/app']} initialIndex={1}>
        <PrivateRoute
          claim="users/modify"
          path="/app"
          claims={[]}
          component={RouteComponent}
        />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
