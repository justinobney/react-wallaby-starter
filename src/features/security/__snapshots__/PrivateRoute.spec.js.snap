// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PrivateRoute should render component when user exists 1`] = `<div />`;

exports[`Component: PrivateRoute should render null when user does not exist 1`] = `null`;
